### PR TITLE
Arborescence

### PR DESCRIPTION
**Pull request by Kzamania**
Ready to close#17 !

### CODE REVIEW

File: e2e/src/areas-arborescence.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+describe ('Empty arboresence ', function() {
+    let page: AreasArborescence;
+
+    beforeEach(() => {
+        page = new AreasArborescence();
+    });
+
+
+      /**
```
Comment (vincentfougeras) [inline e2e/src/areas-arborescence.e2e-spec.ts]: Fix indentation

File: e2e/src/areas-arborescence.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+       *Show an empty arborescence
```
Comment (vincentfougeras) [inline e2e/src/areas-arborescence.e2e-spec.ts]: Add a space after the *

File: src/app/app.component.ts [hunk 3]
```diff
@@ -3,7 +3,6 @@ import { Component } from '@angular/core';
 import { Platform } from '@ionic/angular';
 import { SplashScreen } from '@ionic-native/splash-screen/ngx';
 import { StatusBar } from '@ionic-native/status-bar/ngx';
-
```
Comment (vincentfougeras) [inline src/app/app.component.ts]: Don't remove this line

File: src/app/area/area-db.service.ts [hunk 4]
```diff
@@ -18,15 +19,26 @@ export class AreaDbService {
...
     });
     this.storage.get('type').then((types) => {
       if (types === null) {
         this.storage.set('type', []);
+        // this.initTestType();
       }
     });
   }
 
+  initTestType() {
```
Comment (vincentfougeras) [inline src/app/area/area-db.service.ts]: Add JSDoc

File: src/app/area/area-db.service.ts [hunk 5]
```diff
@@ -18,15 +19,26 @@ export class AreaDbService {
...
+    this.storage.set('type', data.Types);
+  }
+
+  initTestArea() {
```
Comment (vincentfougeras) [inline src/app/area/area-db.service.ts]: Add JSDoc

File: src/app/area/areas-arborescence/areas-arborescence.page.html [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      <ng-container *ngIf="parentArea; else noArea">
+          <ion-title>{{ parentArea.name }}</ion-title>
+      </ng-container>
+      <ng-template #noArea>
+        <ion-title>{{ title }}</ion-title>
+      </ng-template>
+
+
+      <ion-buttons slot="end">
+        <ion-button  (clic)="clicValid()" size="large">
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.html]: Change `(clic)` to `(click)`. Also rename the function to something more explicit, like optionsClicked

File: src/app/area/areas-arborescence/areas-arborescence.page.html [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    <ng-container *ngIf="childAreas && childAreas.length; else noChild">
+      <ion-list>
+        <div *ngFor="let item of childAreas">
+        <ion-card>
+          <ion-item [routerLink]="'/tabs/tab2/area/list/' + item.id " class="arealist">
+            <ion-card-header>
+              <ion-card-title>{{item.name}}</ion-card-title>
+            </ion-card-header>
+            <!-- add a clickable button -->
+            <ion-button href='##' (click)="clicValid()" slot="end" class="listebuttons">
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.html]: Change `listebuttons` to something more explicit, like `ellipsis-button`

File: src/app/area/areas-arborescence/areas-arborescence.page.html [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+              <ion-icon name="more"></ion-icon>
+            </ion-button>
+          </ion-item>
+        </ion-card>
+      </div>
+    </ion-list>
+    </ng-container>
+
+    <ng-template #noChild>
+      <p>pas trouvé</p>
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.html]: Add a more explicit error message

File: src/app/area/areas-arborescence/areas-arborescence.page.scss [hunk 9]
```diff
@@ -0,0 +1,37 @@
+ //--------Arborescence des espaces--------\\
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.scss]: In English please ;)

File: src/app/area/areas-arborescence/areas-arborescence.page.scss [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+  //bouton : des choix dans areas arboresence
+  .listebuttons{
+    --background:var(--ion-color-tertiary);
+    --box-shadow:unset;
+    font-size:14px;
+    --color:var(--ion-color-tertiary-contrast);
+    margin-inline-end:-16px;
+    height:40px;    
+  }
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.scss]: Add missing newline

File: src/app/area/areas-arborescence/areas-arborescence.page.spec.ts [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+});
+
+describe('AreasArborescencePage with parentId', () => {
+  let component: AreasArborescencePage;
+  let fixture: ComponentFixture<AreasArborescencePage>;
+
+  let getAreaByIdSpy: any, getChildAreaByIdSpy: any, getRootAreaSpy: any;
+
+  const parentId = '1';//data.Areas[0].id; // 1
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.spec.ts]: Remove comment

File: src/app/area/areas-arborescence/areas-arborescence.page.spec.ts [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+  const area = data.Areas[0];
+  const area2 = data.Areas[1];
+  const child = data.Areas[2];
+//    const area = {
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.spec.ts]: Remove comment

File: src/app/area/areas-arborescence/areas-arborescence.page.ts [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+})
+export class AreasArborescencePage implements OnInit {
+
+  childAreas: Area[] ;
+  parentArea: Area ;
+  title: String = 'Arborescence';
+  parentId: number;
+  init = true;
+
+  /**
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.ts]: Remove this JSDoc

File: src/app/area/areas-arborescence/areas-arborescence.page.ts [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+@Component({
+  selector: 'app-areas-arborescence',
+  templateUrl: './areas-arborescence.page.html',
+  styleUrls: ['./areas-arborescence.page.scss'],
+})
+export class AreasArborescencePage implements OnInit {
+
+  childAreas: Area[] ;
+  parentArea: Area ;
+  title: String = 'Arborescence';
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.ts]: Initialize attributes in the constructor

File: src/app/area/areas-arborescence/areas-arborescence.page.ts [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+  parentId: number;
+  init = true;
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.ts]: Explain (using JSDoc) what init is for

File: src/app/area/areas-arborescence/areas-arborescence.page.ts [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+  /**
+   *
+   * @param areaDBService : the service
+   * @param route
+   */
+  constructor(private areaDBService: AreaDbService, private route: ActivatedRoute) {}
+
+  ngOnInit() {
+    if (this.init) {
+      // console.log("init");
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.ts]: Remove console.log comments here and below

File: src/app/area/areas-arborescence/areas-arborescence.page.ts [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+      this.checkAreas();
+    }
+  }
+
+  private checkAreas() {
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.ts]: Add missing JSDoc

File: src/app/area/areas-arborescence/areas-arborescence.page.ts [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+  ionViewDidEnter() {
+    if (!this.init) {
+      // console.log("ionEnter");
+      this.checkAreas();
+    } else {
+      this.init = !this.init;
+    }
+  }
+
+  /**
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.ts]: Remove this comment

File: src/app/model/area.ts [hunk 19]
```diff
@@ -9,4 +9,4 @@ export class Area {
 export class Type {
     id: Number;
     name: String;
-}
```
Comment (vincentfougeras) [inline src/app/model/area.ts]: Revert your changes to this file (restore missing newline)

File: src/theme/variables.scss [hunk 20]
```diff
@@ -84,10 +83,10 @@
 
   /**=================================================================**/
   /**------------------Modifications du thème ------------------------**/
-
   ion-tab-bar {
     border-color : var(--ion-color-primary);
-    height : 48px;
+    height : 52px;
+
```
Comment (vincentfougeras) [inline src/theme/variables.scss]: Remove useless newline

File: src/theme/variables.scss [hunk 21]
```diff
@@ -114,21 +114,34 @@
...
+    --padding-start : 12px;
+    --padding-top : 12px;
+    --padding-bottom : 12px;
+    font-size: 18px;
   }
 
+  ion-fab-button{
+    --background:var(--ion-color-secondary);
+    --color:var(--ion-color-secondary-contrast);
+    //margin-right:50px;
```
Comment (vincentfougeras) [inline src/theme/variables.scss]: Remove comment

File: src/theme/variables.scss [hunk 22]
```diff
@@ -114,21 +114,34 @@
...
+  }
+
+
+ 
+
+
```
Comment (vincentfougeras) [inline src/theme/variables.scss]: Remove these newlines

File: src/app/area/areas-arborescence/areas-arborescence.page.html (hunk 6 above)
Comment (Kzamania) [reply src/app/area/areas-arborescence/areas-arborescence.page.html]: What is `(click). ?`

File: src/app/area/areas-arborescence/areas-arborescence.page.html [hunk 23]
```diff
@@ -13,7 +13,7 @@
 
 
       <ion-buttons slot="end">
-        <ion-button  (clic)="clicValid()" size="large">
+        <ion-button  (clic) ="optionsClicked()" size="large">
```
Comment (vincentfougeras) [inline src/app/area/areas-arborescence/areas-arborescence.page.html]: `(clic)` is not a valid Angular event. `(click)` is the valid DOM event that you should use here.
Also, please remove the space that you added before the `=`

**Pull request merged**